### DISCUSSION TITLE
BOの初期構造

### DISCUSSION THREAD

**Discussion by nim-hrkn** [Tomoki-YAMASHITA/CrySPY#33] [General]
下のスクリプトだとBOで初期構造を5作って、その後5構造づつ構造最適化してBOます。
BOで初期構造を10作って、その後2構造づつ評価していく場合はどう書くのでしょうか。
<pre>
[basic]
algo = BO
calc_code = LAMMPS
tot_struc = 30
nstage = 1
njob = 5
jobcmd = bash
jobfile = job_cryspy

[structure]
natot = 8
atype = Ga N
nat = 4 4

[LAMMPS]
lammps_infile = in.lammps
lammps_outfile = out.lammps
lammps_data = structure.data
lammps_potential = GaN.tersoff

[BO]
nselect_bo = 5
dscrpt = FP
score = TS
fp_rmin = 0.5
fp_rmax = 5.0
fp_npoints = 20
fp_sigma = 1.0

[option]
</pre>

**Comment by Tomoki-YAMASHITA** [comment]
初期構造は tot_struc ，n構造ずつBOで選択したい場合は nselect_bo = 2 にします．

使ってないかと思いますが，aiidaではなく通常使用の場合，njobはこの場合2以上にしておけば2個ずつ最適化のジョブを投入します．

```
[basic]
algo = BO
calc_code = LAMMPS
tot_struc = 10
nstage = 1
njob = 5
jobcmd = bash
jobfile = job_cryspy

[structure]
natot = 8
atype = Ga N
nat = 4 4

[LAMMPS]
lammps_infile = in.lammps
lammps_outfile = out.lammps
lammps_data = structure.data
lammps_potential = GaN.tersoff

[BO]
nselect_bo = 2
dscrpt = FP
score = TS
fp_rmin = 0.5
fp_rmax = 5.0
fp_npoints = 20
fp_sigma = 1.0

[option]
```

**Comment by nim-hrkn** [comment]
質問の書き方が悪かったので質問し直します。
サンプルサイズ100（tot_struc = 100)として
一回目は5サンプルでは回帰がうまくできないので、10サンプル程度の評価をしてBO。
二回目は5サンプルを追加評価して15サンプルでBO
三回目はさらに5サンプルを追加評価して20サンプルでBO
という場合の書き方です。
一回目はnselect_bo  = 10
二回目からはnselect_bo = 5と書き直すのでしょうか。

**Reply by Tomoki-YAMASHITA** [reply]
はい，それならそのやり方でOKです．
1回目は nselect_bo = 10 にしておき，2回目の選択が発動する前に nselect_bo = 5 に直せば，
10構造を訓練して，新しい5構造を選択します．
